app/models/article.rb

```diff
@@ -264,14 +264,20 @@ def self.trigger_index(record, remove)
     # on destroy an article is removed from index in a before_destroy callback #before_destroy_actions
     return if remove
 
-    AlgoliaSearch::AlgoliaJob.perform_later(record, "index_or_remove_from_index_where_appropriate")
+    if record.published && record.tag_list.exclude?("hiring")
+      Search::IndexJob.perform_later("Article", record.id)
+    else
+      Search::RemoveFromIndexJob.perform_later(Article.algolia_index_name, record.id)
+      Search::RemoveFromIndexJob.perform_later("searchables_#{Rails.env}", record.index_id)
```

Comment:
I nearly inlined `index_or_remove_from_index_where_appropriate` and related methods here. There's a slight change in behavior (scheduling 3 jobs instead of one which shouldn't be an issue in this case in my opinion).
Methods used in the `index_or_remove_from_index_where_appropriate` (`remove_algolia_index`, `remove_related_objects`) are also used for synchronous indexing but there is an opportunity for refactoring in the future.